packages/form-plugin/src/directive.ts

```diff
@@ -27,102 +38,95 @@ export class FormDirective implements OnInit, OnDestroy {
+            value: this.form.getRawValue()
           }),
           new UpdateFormStatus({
             path: this.path,
-            status: this._formGroupDirective.form.status
+            status: this.form.status
           }),
           new UpdateFormDirty({
             path: this.path,
-            dirty: this._formGroupDirective.form.dirty
+            dirty: this.form.dirty
           })
         ]);
       });
 
-    this._store
-      .select(state => getValue(state, `${this.path}.dirty`))
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(dirty => {
-        if (this._formGroupDirective.form.dirty !== dirty) {
-          if (dirty === true) {
-            this._formGroupDirective.form.markAsDirty();
-            this._cd.markForCheck();
-          } else if (dirty === false) {
-            this._formGroupDirective.form.markAsPristine();
-            this._cd.markForCheck();
-          }
-        }
-      });
+    this.getStateStream(`${this.path}.dirty`).subscribe(dirty => {
+      if (this.form.dirty === dirty || typeof dirty !== 'boolean') {
+        return;
+      }
 
-    this._store
-      .select(state => getValue(state, `${this.path}.disabled`))
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(disabled => {
-        if (this._formGroupDirective.form.disabled !== disabled) {
-          if (disabled === true) {
-            this._formGroupDirective.form.disable();
-            this._cd.markForCheck();
-          } else if (disabled === false) {
-            this._formGroupDirective.form.enable();
-            this._cd.markForCheck();
-          }
-        }
-      });
+      if (dirty) {
+        this.form.markAsDirty();
+      } else {
+        this.form.markAsPristine();
+      }
+
+      this._cd.markForCheck();
+    });
+
+    this.getStateStream(`${this.path}.disabled`).subscribe(disabled => {
+      if (this.form.disabled === disabled || typeof disabled !== 'boolean') {
+        return;
+      }
+
+      if (disabled) {
+        this.form.disable();
+      } else {
+        this.form.enable();
+      }
+
+      this._cd.markForCheck();
+    });
 
     this._formGroupDirective
       .valueChanges!.pipe(
         debounceTime(this.debounce),
+        distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b)),
+        mergeMap(() => {
+          const value = this._formGroupDirective.control.getRawValue();
+          this._updating = true;
+          return this._store
+            .dispatch([
+              new UpdateFormValue({
+                path: this.path,
+                value
+              }),
+              new UpdateFormDirty({
+                path: this.path,
+                dirty: this._formGroupDirective.dirty
+              }),
+              new UpdateFormErrors({
+                path: this.path,
+                errors: this._formGroupDirective.errors
+              })
+            ])
+            .pipe(finalize(() => (this._updating = false)));
+        }),
```

Comment:
I think that the previous `subscribe` version (as opposed to the `mergeMap` and `finalize`) is clearer, simpler code with less rxjs dependencies.

Reply:
What is the difference, how many dependencies do you have? If they are still under the Angular hood and somehow get into the bundle...

Savings on matches, against readable code